tests/test_centrality.py

```diff
@@ -0,0 +1,113 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestCentralityDiGraph(unittest.TestCase):
+    def setUp(self):
+        self.graph = retworkx.PyDiGraph()
+        self.a = self.graph.add_node("A")
+        self.b = self.graph.add_node("B")
+        self.c = self.graph.add_node("C")
+        self.d = self.graph.add_node("D")
+        edge_list = [
+            (self.a, self.b, 1),
+            (self.b, self.c, 1),
+            (self.c, self.d, 1),
+        ]
+        self.graph.add_edges_from(edge_list)
+
+    def test_betweenness_centrality(self):
+        betweenness = retworkx.digraph_betweenness_centrality(self.graph)
+        expected = {0: 0.0, 1: 0.3333333333333333, 2: 0.3333333333333333, 3: 0.0}
+        self.assertEqual(expected, betweenness)
+
+    def test_betweenness_centrality_endpoints(self):
+        betweenness = retworkx.digraph_betweenness_centrality(
+            self.graph, endpoints=True)
+        expected = {0: 0.25, 1: 0.41666666666666663, 2: 0.41666666666666663, 3: 0.25}
+        self.assertEqual(expected, betweenness)
+
+    def test_betweenness_centrality_unnormalized(self):
+        betweenness = retworkx.digraph_betweenness_centrality(
+            self.graph, endpoints=False, normalized=False)
+        expected = {0: 0.0, 1: 2.0, 2: 2.0, 3: 0.0}
+        self.assertEqual(expected, betweenness)
+
+
+class TestCentralityGraph(unittest.TestCase):
+    def setUp(self):
+        self.graph = retworkx.PyGraph()
+        self.a = self.graph.add_node("A")
+        self.b = self.graph.add_node("B")
+        self.c = self.graph.add_node("C")
+        self.d = self.graph.add_node("D")
+        edge_list = [
+            (self.a, self.b, 1),
+            (self.b, self.c, 1),
+            (self.c, self.d, 1),
+        ]
+        self.graph.add_edges_from(edge_list)
+
+    def test_betweenness_centrality(self):
+        betweenness = retworkx.graph_betweenness_centrality(self.graph)
+        expected = {0: 0.0, 1: 0.6666666666666666, 2: 0.6666666666666666, 3: 0.0}
+        self.assertEqual(expected, betweenness)
+
+    def test_betweenness_centrality_endpoints(self):
+        betweenness = retworkx.graph_betweenness_centrality(
+            self.graph, endpoints=True)
+        expected = {0: 0.5, 1: 0.8333333333333333, 2: 0.8333333333333333, 3: 0.5}
+        self.assertEqual(expected, betweenness)
+
+    def test_betweenness_centrality_unnormalized(self):
+        betweenness = retworkx.graph_betweenness_centrality(
+            self.graph, endpoints=False, normalized=False)
+        expected = {0: 0.0, 1: 2.0, 2: 2.0, 3: 0.0}
+        self.assertEqual(expected, betweenness)
```

Comment:
We've moved to splitting the test files up by graph class they work with. So for this do you mind moving it to a new file `tests/graph/test_centrality.py` and move the remaining digraph cases in this file to `tests/digraph/test_centrality.py`? (we probably should update the contributing guide to outline this test dir structure it was an oversight in #296 that we didn't do that then)

Also do you want to add a test class for `PyGraph` with holes too to match up  the coverage for `PyDiGraph` below?

Reply:
I don't know if you saw my comment above: I'm concerned about the possibility that someone creates a graph with, say, five nodes, but the indices are all greater than `10^6`. In the current implementation, the algorithm would create and iterate over vecs of length greater than `10^6`. I wonder if this is some that is possible and if so should we account for it, i.e. give a more efficient implementation.

Reply:
Oh I must have missed that comment. It is technically possible but a pretty pathological edge case. Mostly because stable graph maintains a list of free nodes and reuses indices on new nodes after removal. So to get this state you'd have to create a graph with a million nodes, then delete all but 5, and then run centrality on it. We probably could use a hashmap or btreemap instead of a vec and then we'd only iterate over n nodes, but the tradeoff there is the iteration performance is lower than a vec. Personally I'm fine either way and don't have a strong preference.